heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCache.java

```diff
@@ -0,0 +1,187 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.twitter.heron.common.config.SystemConfig;
+import com.twitter.heron.metricsmgr.MetricsSinksConfig;
+import com.twitter.heron.proto.system.Common;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+import com.twitter.heron.spi.metricsmgr.metrics.MetricsFilter;
+
+/**
+ * compatible with tmaster interface
+ * see heron/tmaster/src/cpp/manager/tmetrics-collector.h
+ */
+public class MetricsCache {
+  public static final String METRICS_SINKS_TMASTER_SINK = "tmaster-sink";
+  public static final String METRICS_SINKS_TMASTER_METRICS = "tmaster-metrics-type";
+
+  // logger
+  private static final Logger LOG = Logger.getLogger(CacheCore.class.getName());
+
+  private CacheCore cache = null;
+  private MetricsFilter metricNameType = null;
+
+  /**
+   * constructor
+   *
+   * @param systemConfig heron config
+   * @param sinksConfig sink config
+   */
+  public MetricsCache(SystemConfig systemConfig, MetricsSinksConfig sinksConfig) {
+    // metadata
+    metricNameType = new MetricsFilter();
+    Map<String, Object> sinksTmaster = sinksConfig.getConfigForSink(METRICS_SINKS_TMASTER_SINK);
+    @SuppressWarnings("unchecked")
+    Map<String, String> metricsType =
+        (Map<String, String>) sinksTmaster.get(METRICS_SINKS_TMASTER_METRICS);
+    for (Map.Entry<String, String> e : metricsType.entrySet()) {
+      metricNameType.setMetricToType(e.getKey(), TranslateFromString(e.getValue()));
+    }
+    //
+    long maxInterval = systemConfig.getTmasterMetricsCollectorMaximumIntervalMin() * 60;
+    long interval = systemConfig.getTmasterMetricsCollectorPurgeIntervalSec();
+    long maxException = systemConfig.getTmasterMetricsCollectorMaximumException();
+
+    cache = new CacheCore(maxInterval, interval, maxException);
+
+    cache.startPurge();
+  }
+
+  private MetricsFilter.MetricAggregationType TranslateFromString(String type) {
+    if ("SUM".equals(type)) {
+      return MetricsFilter.MetricAggregationType.SUM;
+    } else if ("AVG".equals(type)) {
+      return MetricsFilter.MetricAggregationType.AVG;
+    } else if ("LAST".equals(type)) {
+      return MetricsFilter.MetricAggregationType.LAST;
+    } else {
+      LOG.log(Level.SEVERE, "Unknown metrics type in metrics sinks " + type);
+      return MetricsFilter.MetricAggregationType.UNKNOWN;
+    }
+  }
+
+  /**
+   * sink publishes metrics and exceptions to this interface
+   *
+   * @param metrics message from sinks
+   */
+  public void AddMetric(TopologyMaster.PublishMetrics metrics) {
```

Comment:
Method names should start with lowercase (e.g., addMetric) here and below in this class and in others (e.g., MetricsCacheQueryUtils). Also if it takes multiple metrics it should have an s at the end (e.g. addMetrics).